single-line-diagram-core/src/main/java/com/powsybl/sld/layout/VerticalSubstationLayout.java

```diff
@@ -69,19 +69,26 @@ protected Coord calculateCoordVoltageLevel(LayoutParameters layoutParam, Graph v
                         infosNbSnakeLines.getNbSnakeLinesBottomVL().compute(node1.getGraph().getVoltageLevelInfos().getId(), (k, v) -> v + 1);
                         infosNbSnakeLines.getNbSnakeLinesBottomVL().compute(node2.getGraph().getVoltageLevelInfos().getId(), (k, v) -> v + 1);
                     }
-                    infosNbSnakeLines.getNbSnakeLinesLeftRight().compute(Side.RIGHT, (k, v) -> v + 1);
                     double decal1V = infosNbSnakeLines.getNbSnakeLinesBottomVL().get(node1.getGraph().getVoltageLevelInfos().getId()) * layoutParam.getVerticalSnakeLinePadding();
                     double decal2V = infosNbSnakeLines.getNbSnakeLinesBottomVL().get(node2.getGraph().getVoltageLevelInfos().getId()) * layoutParam.getVerticalSnakeLinePadding();
-                    double xSnakeLine = maxH + infosNbSnakeLines.getNbSnakeLinesLeftRight().get(Side.RIGHT) * layoutParam.getHorizontalSnakeLinePadding();
-
-                    pol.addAll(Arrays.asList(x1, y1,
-                            x1, initY1 + decal1V,
-                            xSnakeLine, initY1 + decal1V,
-                            xSnakeLine, initY2 + decal2V,
-                            x2, initY2 + decal2V,
-                            x2, y2));
+
+                    if (node1.getGraph() == node2.getGraph()) { // in the same VL (horizontal layout)
```

Comment:
Currently we only have one layout per diagram and the calculation of the snakelines must be done globally because the `infosNbSnakeLines` management must be common.
The horizontal version of calculatePolylineSnakeLine() is only valid for a `VoltageLevelLayout` or a `HorizontalSubstationLayout`.
For a `VerticalSubstationLayout`, we have to mix.
This may be due to a design problem.

Reply:
I thought that as InfosNbSnakeLines was used in both cases it could handle it, but in fact due to a design problem it doesn't indeed. It's because there are three maps in `InfosNbSnakeLines` used for horizontal or vertical, instead of a single one for both cases. Let's forget about this for this PR.

Reply:
Ok